src/server/es/index.js

```diff
@@ -53,14 +53,20 @@ class ES {
       },
     };
     validatedQueryBody.track_total_hits = true;
-    log.info('[ES.query] index, type, query body: ', esIndex, esType, JSON.stringify(validatedQueryBody));
-    return this.client.search({
+
+    var start = new Date().getTime();
+    var searchResult = this.client.search({
       index: esIndex,
       body: validatedQueryBody,
     }).then((resp) => resp.body, (err) => {
       log.error(`[ES.query] error during querying: ${err.message}`);
       throw new Error(err.message);
     });
+    var end = new Date().getTime();
```

Comment:
this.client.search is async so this end time won't reflect when the request ends, just when it was sent. 

you'll want to add a call to finally after the .then and add this and the code below.

Reply:
Good catch, I'll fix it. Is it obvious I don't with Javascript very often? :laughing:

Reply:
something like this 

```
var searchResult = this.client.search({
  index: esIndex,
  body: validatedQueryBody,
}).then((resp) => resp.body, (err) => {
  log.error(`[ES.query] error during querying: ${err.message}`);
  throw new Error(err.message);
}).finally(() => {
   var end = new Date().getTime();
   var durationInMS = end - start;

   log.info('[ES.query] DurationInMS:' + durationInMS + '. index, type, query body: ', esIndex, esType, JSON.stringify(validatedQueryBody));
   return searchResult;
})
```

and no worries i almost commented on the log.info asking you to use f strings before realizing this was javascript